### PR TITLE
feat: Use global tokio runtime per executor

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

N/A

## Rationale for this change

This was an experiment related to researching https://github.com/apache/datafusion-comet/issues/1098. I wondered if creating a new tokio runtime per task was causing overhead. It does not seem to matter too much but this PR does result in a small performance improvement for my benchmarking (2% faster for TPC-H @ 100 GB).

## What changes are included in this PR?

Create one global tokion runtime for the process.

### CODE REVIEW

**Comment by kazuyukitanimura** [comment]
Wondering if the performance improvement comes from different numbers of threads? Are the default numbers are the same?

File: native/core/src/execution/jni_api.rs [hunk 1]
```diff
@@ -61,6 +61,20 @@ use tokio::runtime::Runtime;
 use crate::execution::operators::ScanExec;
 use log::info;
 
+use once_cell::sync::Lazy;
+
+static TOKIO_RUNTIME: Lazy<Runtime> = Lazy::new(|| {
```
Comment (parthchandra) [inline native/core/src/execution/jni_api.rs:66]: Perhaps this should be once per session (a user could modify the config in the session or run different queries in a single job that might require independent configuration)?

**Comment by andygrove** [comment]
> Wondering if the performance improvement comes from different numbers of threads? Are the default numbers are the same?

Tokio's defaults are to create one worker thread per core for async code (which can be overridden with the `TOKIO_WORKER_THREADS` env var) and an upper limit of 512 blocking threads (but they are created on demand, so they are unlikely to get this high).

I didn't make a note of the benchmark settings I used but it was likely 8 cores in 1 executor so that this PR would have changed worker threads from 32 to 8.

**Comment by kazuyukitanimura** [comment]
> I didn't make a note of the benchmark settings I used but it was likely 8 cores in 1 executor so that this PR would have changed worker threads from 32 to 8.

If `COMET_WORKER_THREADS` is set to 1 with the existing local tokio runtime, do we still see the 2% gain? If yes, then we can say that is due to the reduced (less congested) threads. If no, we can say it is thanks to the global runtime?

**Comment by andygrove** [comment]
I ran more benchmarks and experimented with number of threads but I am seeing some variability in results. It is hard to tell how much difference this really makes. 

Maybe I should just close this PR, but typically, there is one Tokio runtime per process, and I am not sure what the original reason was for creating a new runtime per native query. @viirya do you know why we do this?

**Comment by viirya** [comment]
> Maybe I should just close this PR, but typically, there is one Tokio runtime per process, and I am not sure what the original reason was for creating a new runtime per native query. @viirya do you know why we do this?

No special reason. The original idea is to create Tokio runtime once going to run a query. And I think that one runtime per query can prevent one particular query to monopolize the runtime and starve another queries?

**Comment by andygrove** [comment]
I'll close this for now. Thanks for the feedback @kazuyukitanimura and @viirya

**Pull request closed without merging**